server/channels/app/scheduled_post_job.go

```diff
@@ -23,6 +23,16 @@ func (a *App) ProcessScheduledPosts(rctx request.CTX) {
 	rctx = rctx.WithLogger(rctx.Logger().With(mlog.String("component", "scheduled_post_job")))
 	rctx.Logger().Debug("ProcessScheduledPosts called...")
 
+	if !*a.Config().ServiceSettings.ScheduledPosts {
+		rctx.Logger().Debug("ProcessScheduledPosts exiting as the feature is turned off via ServiceSettings.ScheduledPosts setting...")
+		return
+	}
+
+	if a.License() == nil {
+		rctx.Logger().Debug("ProcessScheduledPosts exiting as no license is available")
+		return
+	}
+
```

Comment:
I have two questions around this:
1. If I am reading things right, even if you don't have a license, the job will run (and print the debug of `ProcessSchedueldPosts called...` and `ProcessScheudledPost exiting as no license is available`. Is that desired? Shouldn't we just stop the job? 0/5
2. Let's say I have a license. I send 10 scheduled posts for the next 10 days. I lose my license.
  2.1. Do I have any way to find / recover / delete those scheduled posts while I don't have a license? (apart of handling manually the database)
  2.2. 10 days later, I add a new license. Will the 10 messages I scheduled be sent now, all at the same time? Is that expected/desired?

Reply:
1. Yeah I started by adding a license and config change listener but the code was becoming a bit ugly as license change listener needs to check for config and config change listener needs to check for license as well, so essentially we'd be doing the same checks but thrice -once in config change listener and once in license change listener and once in cluster master change listener, so instead of doing all that, I left the job always running and added two simple, easy to read checks. The logs are debug logs so shouldn't show up all the time for users.

2.1 No, there is no way, you loose your scheduled posts just like you loose them if you leave a team or channel.

2.2 - currently the code processes all pending scheduled posts but there is a ticket to restrict this to posts pending the last 24 hours. There would be a spike in DB usage when the feature is enabled but I'll do some performance testing to see how much is that spike and will add some delays between batches to mitigate that.

Reply:
thanks for the explanation, had same concern as @larkox but after reading your explanation, I agree with @harshilsharma63 approach, also do not think is worth the effort + adding such complexity for this edge case scenario.

Reply:
Re: 2.1 and 2.2.
From a purely UX standpoint, Is this the desired behavior? Both "missing the scheduled posts" (even when leaving a channel) and posts being send "out of time or not at all" seems off.

But if it has been discussed and decided, I am fine with it.

Reply:
It was discussed and was part of agreed upon UX.